pkg/pillar/types/clustertypes.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"net"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+const (
+	ClusterStatusPort = "12346" // ClusterStatusPort - Port for k3s server for cluster status advertise
+)
+
+// EdgeNodeClusterConfig - Configuration for cluster multi-node from controller
+type EdgeNodeClusterConfig struct {
+	ClusterName      string
+	ClusterID        UUIDandVersion
+	ClusterInterface string
+	ClusterIPPrefix  *net.IPNet
+	IsWorkerNode     bool
+	JoinServerIP     net.IP
+	BootstrapNode    bool
+
+	// CipherBlockStatus, for encrypted cluster token data
+	CipherToken CipherBlockStatus
+}
+
+// ENClusterAppStatus - Status of an App Instance in the multi-node cluster
+type ENClusterAppStatus struct {
+	AppUUID             uuid.UUID // UUID of the appinstance
+	IsDNSet             bool      // DesignatedNodeID is set for this node
+	ScheduledOnThisNode bool      // App is running on this device
+	StatusRunning       bool      // Status of the app in "Running" state
+	IsVolumeDetached    bool      // Are volumes detached after failover ?
+}
+
+// Key - returns the key for the config of EdgeNodeClusterConfig
+func (config EdgeNodeClusterConfig) Key() string {
+	return config.ClusterID.UUID.String()
+}
+
+// EdgeNodeClusterStatus - Status of the multi-node cluster published by zedkube
+type EdgeNodeClusterStatus struct {
+	ClusterName      string
+	ClusterID        UUIDandVersion
+	ClusterInterface string
```

Comment:
Add comment that this is a logical label of the port.

Reply:
Ok.